taraxa/state/dpos/precompiled/undelegations.go

```diff
@@ -0,0 +1,100 @@
+package dpos
+
+import (
+	"math/big"
+
+	"github.com/Taraxa-project/taraxa-evm/common"
+	"github.com/Taraxa-project/taraxa-evm/core/types"
+	"github.com/Taraxa-project/taraxa-evm/rlp"
+)
+
+type Undelegation struct {
+	// Amount of TARA that accound should be able to get
+	Amount *big.Int
+
+	// Block number when the withdrawal be ready
+	Block types.BlockNum
+
+	// TODO: we will needed it for slashing
+	Validator *common.Address
+}
+
+type Undelegations struct {
+	storage *StorageWrapper
+	// <delegator addres -> list of undelegations> as each delegator can undelegate multiple times
+	undelegations_map map[common.Address]*IterableMap
```

Comment:
Why do you need IterableMap for undelegations ? You dont need to iterate over them do you ?